spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -726,6 +726,53 @@ struct ProductionVisitor
     // Returns a boolean expression that's 'true' if a 'stop' was encountered.
     Expression parseProduction(const Production& p) { return _parseProduction(p, p.meta()); }
 
+    void skipRegExp(const hilti::ctor::RegExp& c) {
+        auto re = ID("__re");
+        int i = 0;
+        while ( pb->cg()->haveAddedDeclaration(re) )
+            re = ID(fmt("__re_%" PRId64, ++i));
```

Comment:
Hmm ... Feels like we should have a codegen method here that (1) takes care of creating a unique ID, and (2) checks if the same constant already exists and reuses that instead to avoid duplication. it's ok to leave as is for now, but add comment at least to that regard I'd say.